fix: only authenticated permission on the api

asadazam93 opened the pull request:
[TNL-9412](https://openedx.atlassian.net/browse/TNL-9412)

Since the api is used to render the tab it only needs to have the authenticated permission

awaisdar001 reviewed (changes requested):
Can you add a test case for this scenrio?

awaisdar001 commented on lms/djangoapps/learner_dashboard/tests/test_views.py:
```diff
@@ -93,10 +94,15 @@ def test_program_discussion_disabled(self):
+        Test if API returns iframe in case ProgramDiscussionsConfiguration model contains proper data for staff
+        and non staff. In case of non staff user must be be enrolled in program.
         """
+        if staff:
+            self.user = UserFactory(is_staff=True)
+            self.client.login(username=self.user.username, password=self.password)
```
`self.user = UserFactory(is_staff=staff)`

asadazam93 replied:
This will fail because it will create a new user with no enrolment in case of non staff user

awaisdar001 commented on lms/djangoapps/learner_dashboard/tests/test_views.py:
```diff
@@ -93,10 +94,15 @@ def test_program_discussion_disabled(self):
+    def test_api_returns_discussions_iframe(self, staff):
         """
-        Test if API returns iframe in case ProgramDiscussionsConfiguration model contains proper data
+        Test if API returns iframe in case ProgramDiscussionsConfiguration model contains proper data for staff
+        and non staff. In case of non staff user must be be enrolled in program.
```
Tests that API returns LTI discussion iframe when the provider is properly configured. The scenario is true for both staff and nonstaff users.

asadazam93 replied:
The scenario is not true for both staff and nonstaff. non staff requires an additional enrolment condition.

awaisdar001 replied:
I see got it. Then I think its simpler to break the test into two? what do you think?

asadazam93 replied:
The current test is simpler. In case of staff user I create a new user that no enrolment and test with that. The default user is a non staff user with an enrolment

awaisdar001 commented on lms/djangoapps/learner_dashboard/tests/test_views.py:
```diff
@@ -114,18 +120,19 @@ def test_api_returns_discussions_iframe(self):
-    def test_program_access_denied(self):
-        """
-        Test if API returns 403 in case user has no access to program
+        Test if API returns default response in case user has no access to program
```
This docstring also needs to improve and explain the scenario under test. 

Test that the API does not break if the discussion provider is not configured, instead it should return empty data for discussion iframe.

asadazam93 replied:
This is not the scenario that is being tested here. The current docstring is correct.

awaisdar001 replied:
1) "default response" is misleading here. 
2) please mention what kind of access.

awaisdar001 commented on lms/djangoapps/learner_dashboard/tests/test_views.py:
```diff
@@ -114,18 +120,19 @@ def test_api_returns_discussions_iframe(self):
         """
-        Test if API returns 404 in case program does not exist
-        """
-        response = self.client.get(reverse('program_discussion', kwargs={'program_uuid': str(uuid4())}))
-        self.assertEqual(response.status_code, 404)
```
why this test is removed?

asadazam93 replied:
Because the permission has been removed so this case does not occur anymore

awaisdar001 replied:
got it thanks - resolving this conv.

awaisdar001 commented on lms/djangoapps/learner_dashboard/tests/test_views.py:
```diff
@@ -114,18 +120,19 @@ def test_api_returns_discussions_iframe(self):
         self.assertIn('iframe', str(response.data['discussion']['iframe']), )
 
-    def test_program_does_not_exist(self):
+    def test_program_empty_response(self):
```
`test_program_without_provider_config`

asadazam93 replied:
This is not what is being tested. Testing empty response in case the user does not have permission here

awaisdar001 replied:
permission in case of enrollment? or any other access please include that into the docstring.

awaisdar001 reviewed (changes requested):
SOme changes are requested. Please take a look 

additionally, Please add another test for staff user when there is no discussion provider configuration available for a program.

asadazam93 commented:
> SOme changes are requested. Please take a look
> 
> additionally, Please add another test for staff user when there is no discussion provider configuration available for a program.

Addressed all comments. No changes needed. The scenario when there is no discussion provider configuration available is the same for both staff and non staff and it is already covered in a test case

saadyousafarbi reviewed (approved):
Looks good!
